tests/component-tests/uiOptions/js/UIOptionsTests.js

```diff
@@ -15,30 +15,150 @@ https://github.com/fluid-project/infusion/raw/master/Infusion-LICENSE.txt
 (function ($) {
     "use strict";
 
-    $(document).ready(function () {
+    var customizedTocTemplate = "../../../../src/components/tableOfContents/html/TableOfContents.html";
+
+    /* Mixin grade for UIO test component */
+    fluid.defaults("fluid.uiOptions.testPrefsEditorBase", {
+        gradeNames: ["fluid.uiOptions.prefsEditor"],
+        terms: {
+            templatePrefix: "../../../../src/framework/preferences/html",
+            messagePrefix: "../../../../src/framework/preferences/messages"
+        }
+    });
+
+    /* Mixin grade for UIO test environment */
+    fluid.defaults("fluid.uiOptions.prefsEditorBaseTest", {
+        gradeNames: ["fluid.test.testEnvironment"],
+        components: {
+            prefsEditor: {
+                createOnEvent: "{prefsEditorTester}.events.onTestCaseStart"
+            },
+            prefsEditorTester: {}
+        }
+    });
 
-        jqUnit.module("UIOptions Tests");
+    fluid.defaults("fluid.uiOptions.testPrefsEditorCustomToc", {
+        gradeNames: ["fluid.uiOptions.testPrefsEditorBase"],
+        tocTemplate: customizedTocTemplate
+    });
+
+    fluid.defaults("fluid.uiOptions.prefsEditorCustomTocTester", {
+        gradeNames: ["fluid.test.testCaseHolder"],
+        modules: [{
+            name: "UIOptions Tests",
+            tests: [{
+                name: "Pass in customized toc template",
+                expect: 2,
+                sequence: [{
+                    event: "{prefsEditorCustomTocTest prefsEditor}.events.onCreate",
+                    listener: "jqUnit.assertEquals",
+                    args: ["The toc template is applied properly to the pageEnhancer", customizedTocTemplate, "{prefsEditor}.enhancer.uiEnhancer.fluid_prefs_enactor_tableOfContents.options.tocTemplate"]
+                },
+                {
+                    funcName: "fluid.identity"
+                },
```

Comment:
I'm assuming this call to `fluid.identity` is to try to get around having the two events be back-to-back. Rather than do it this way, can you just do all the tests after the last event. If you really need to listen and run tests to the different events independently and you are not able to just specify both events, please make sure to leave a comment and link to the relevant JIRA issue for why this is necessary.

Reply:
Also, it appears that this test could just use the plain jqUnit.async test instead. You'd have separate listeners to each event and run jqUnit.start() in the listener that executes last.